packages/bls-snap/src/utils/snapUtils.ts

```diff
@@ -32,24 +27,24 @@ export async function upsertNetwork(
       });
     }
 
-    const storedNetwork = getNetwork(state, network.chainId);
+    const _network = getNetwork(state, network.chainId);
     // eslint-disable-next-line no-negated-condition
-    if (!storedNetwork) {
+    if (!_network) {
       if (!state.networks) {
         state.networks = [];
       }
 
       state.networks.push(network);
     } else {
-      if (JSON.stringify(storedNetwork) === JSON.stringify(network)) {
+      if (JSON.stringify(_network) === JSON.stringify(network)) {
```

Comment:
If the keys/props in `_network` and `network` are not in the same order but they are deeply equal, these `stringify` calls will produce different strings. It would be better to check for deep equality, such as with [_.isEqual](https://lodash.com/docs/4.17.15#isEqual) (or similar) instead.

Reply:
indeed , swapped to `deep-equal`